transport/peer/peerlist/single.go

```diff
@@ -0,0 +1,61 @@
+package peerlist
+
+import (
+	"context"
+
+	"go.uber.org/yarpc/transport"
+	"go.uber.org/yarpc/transport/internal/errors"
+
+	"github.com/uber-go/atomic"
+)
+
+type single struct {
+	peerID  transport.PeerIdentifier
+	peer    transport.Peer
+	agent   transport.PeerAgent
+	started atomic.Bool
+}
+
+// NewSingle creates a static PeerList with a single Peer
+func NewSingle(pi transport.PeerIdentifier, agent transport.PeerAgent) transport.PeerList {
+	return &single{
+		peerID: pi,
+		agent:  agent,
+	}
+}
+
+func (pl *single) Start() error {
+	if pl.started.Swap(true) {
+		return errors.ErrPeerListAlreadyStarted("single")
```

Comment:
I think you want to use a RW Mutex instead. There's a race here where if I
call `Start` and `ChoosePeer` concurrently, `started` is set to true by
`Start` but `ChoosePeer` returns the `nil` peer before `Start` has a chance to
fill it in.